coalib/processes/Processing.py

```diff
@@ -421,6 +429,31 @@ def instantiate_processes(section,
             bear_runner_args)
 
 
+def uncache_if_unloaded_bears(initial_count,
+                              loaded_count,
+                              files_dict,
+                              cache,
+                              log_printer):
+    """
+    Checks if any bear could not be instantiated because of a missing
+    dependency and uncaches current section's cached files
+    to permit a future execution of the failing bear.
+
+    :param initial_count: The number of bears that the section should execute.
+    :param loaded_count:  The number of bears that were actually
+                          loaded with success.
+    :param files_dict:    The files this section runs on. Theese are the files
+                          that will be removed from the cache.
+    :param cache:         An instance of ``misc.Caching.FileCache`` to use as
+                          a file cache buffer.
+    :param log_printer    The log printer to warn to.
+    """
+    if initial_count != loaded_count:
+        log_printer.debug('The catch has been reset because one of '
```

Comment:
it is not always because of one of the bears right?  maybe you should modify this message